### PR TITLE
Adds AccountSet Transaction Tests for `Signer`

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds additional testing for `Signer` for `AccountSet` transactions. 

### Context of Change

This is a part of an initiative to have better testing for `Signer`, to catch bugs quickly and more easily in the future. 

### Type of Change

- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Before / After

New tests added for `Signer`. 

## Test Plan

CI passes. 

## Future Tasks
More tests for `Signer` will be added.

### CODE REVIEW

**Review by tedkalaw — APPROVED** [review]

File: test/XRP/fakes/fake-xrp-protobufs.ts [hunk 1]
```diff
@@ -194,27 +288,58 @@ paymentAllFields.setPathsList(pathList)
...
 /**
  * Helper function to generate Transaction objects with the standard values from Payment objects.
+ * There must be at most one of accountSet or payment.
  *
+ * @param accountSet -AccountSet object to insert into the transaction.
  * @param payment -Payment object to insert into the transaction.
  * @returns Payment Transaction with the included payment param.
  */
-function buildStandardTransactionFromPayment(payment: Payment): Transaction {
+function buildStandardTestTransaction(
```
Comment (amiecorso) [inline test/XRP/fakes/fake-xrp-protobufs.ts:297]: as this grows, another option is: let this take two arguments: first one a `TransactionType` enum as you will find in the xpring-sdk.  the second one the actual transaction data, which can be typed as `TransactionData` with a type alias like `TransactionData = Payment | AccountSet | TrustSet | etc.` (whatever the exact typescript is for this).  Then switch case on the `TransactionType`.

File: test/XRP/signer.test.ts [hunk 2]
```diff
@@ -222,6 +227,119 @@ describe('Signer', function (): void {
...
+    assert.deepEqual(signedTransaction, expectedSignedTransaction)
+  })
+
+  it('Sign AccountSet transaction with no fields', function (): void {
+    // GIVEN a AccountSet transaction with no fields set, a wallet and expected signing artifacts.
+    const wallet = new FakeWallet(fakeSignature)
+
+    // Encode transaction with the expected signature.
+    const expectedSignedTransactionJSON = Serializer.transactionToJSON(
+      testTransactionAccountSetOneField,
```
Comment (amiecorso) [inline test/XRP/signer.test.ts:298]: shouldn't this be no fields?

File: test/XRP/signer.test.ts [hunk 3]
```diff
@@ -222,6 +227,119 @@ describe('Signer', function (): void {
...
+      wallet,
+    )
+
+    // THEN the signing artifacts are as expected.
+    assert.exists(signedTransaction)
+    assert.deepEqual(signedTransaction, expectedSignedTransaction)
+  })
+
+  it('Sign AccountSet transaction with special case fields', function (): void {
+    // GIVEN a AccountSet transaction with fields set with special values, a wallet and expected signing artifacts.
```
Comment (amiecorso) [inline test/XRP/signer.test.ts:321]: whoa special values

**Review by amiecorso — APPROVED** [review]
approving now but take a look at comments because I think there might be a typo in one test

**Pull request merged**